pkg/apiclient/argo-kube-client.go

```diff
@@ -92,7 +93,7 @@ func newArgoKubeClient(ctx context.Context, clientConfig clientcmd.ClientConfig,
 func (a *argoKubeClient) NewWorkflowServiceClient() workflowpkg.WorkflowServiceClient {
 	wfArchive := sqldb.NullWorkflowArchive
 	wfLister := store.NewKubeLister(a.wfClient)
-	return &errorTranslatingWorkflowServiceClient{&argoKubeWorkflowServiceClient{workflowserver.NewWorkflowServer(a.instanceIDService, argoKubeOffloadNodeStatusRepo, wfArchive, a.wfClient, wfLister, nil)}}
+	return &errorTranslatingWorkflowServiceClient{&argoKubeWorkflowServiceClient{workflowserver.NewWorkflowServer(a.instanceIDService, argoKubeOffloadNodeStatusRepo, wfArchive, a.wfClient, wfLister, nil, v1.NamespaceAll)}}
```

Comment:
The SDK doesn't need a namespace since it doesn't use the Reflector/Store, right?

Reply:
yes, you are right. Although it doesn't matter what namespace we pass here, I made it a nil pointer to make it more clear as you suggested.

Reply:
Yea the explicitness is helpful.
Although I forgot that `nil` only works for pointers in Go, and that the string default `""` is, perhaps confusingly, equivalent to cluster-scope, so you had to refactor a bunch here. The other option would be to use an invalid NS name like `"-1"`, but I think `nil` is clearer than that.